eo-maven-plugin/src/test/java/org/eolang/maven/PullMojoIT.java

```diff
@@ -0,0 +1,164 @@
+            f -> {
+                f.clean();
+                f.files().file("src/main/eo/foo.eo").write(
+                    String.join(
+                        "\n",
+                        "+package org.eolang",
+                        "",
+                        "# In this program, we refer to the 'String' object by mistake.",
+                        "[] > foo",
+                        "  \"Hello\" > @"
+                    ).getBytes()
+                );
+                PullMojoIT.configureFarea(f);
+                f.exec("eo:register", "eo:parse", "eo:shake", "eo:probe", "eo:pull");
+                f.files().file("src/main/eo/foo.eo").write(
+                    String.join(
+                        "\n",
+                        "+package org.eolang",
+                        "",
+                        "# Now, this program, doesn't refer to the 'String' object",
+                        "[] > foo",
+                        "  42 > @"
+                    ).getBytes()
+                );
+                f.exec("eo:register", "eo:parse", "eo:shake", "eo:probe", "eo:pull");
+                MatcherAssert.assertThat(
+                    "Foreign must not contain a reference to an old object",
+                    PullMojoIT.ids(temp.resolve("target/eo-foreign.json")),
+                    Matchers.not(Matchers.hasItem("org.eolang.string"))
+                );
+                MatcherAssert.assertThat(
+                    "necessary objects must were pulled, but it didn't",
+                    temp.resolve(
+                        String.format(
+                            "target/eo/%s/org/eolang/number.eo",
+                            PullMojo.DIR
+                        )
+                    ).toFile().exists(),
+                    Matchers.is(true)
+                );
+                MatcherAssert.assertThat(
+                    "unnecessary objects were not removed",
+                    temp.resolve(
+                        String.format(
+                            "target/eo/%s/org/eolang/string.eo",
+                            PullMojo.DIR
+                        )
+                    ).toFile().exists(),
+                    Matchers.is(false)
+                );
+            }
+        );
+    }
+
+    private static void configureFarea(final Farea farea) throws IOException {
+        farea.build()
+            .plugins()
+            .append(
+                "org.eolang",
+                "eo-maven-plugin",
+                System.getProperty(
+                    "eo.version",
+                    Manifests.read("EO-Version")
+                )
+            )
+            .configuration()
+            .set("foreign", "${project.build.directory}/eo-foreign.json")
+            .set("foreignFormat", "json");
+    }
+
+    private static List<String> ids(final Path path) throws IOException {
+        final JsonArray json = Json.createReader(Files.newBufferedReader(path)).readArray();
```

Comment:
@Suban05 I think you can use our `TjsForeign` object here instead this JSON stuff